chainntnfs/txnotifier.go

```diff
@@ -322,6 +335,20 @@ func NewSpendRequest(op *wire.OutPoint, pkScript []byte) (SpendRequest, error) {
 	}
 	r.PkScript = outputScript
 
+	// For Taproot spends we have the main problem that for the key spend
+	// path we cannot derive the pkScript from only looking at the input's
+	// witness. So we need to rely on the outpoint information alone.
+	if outputScript.Class() == txscript.WitnessV1TaprootTy {
+		if op == nil {
+			return r, fmt.Errorf("cannot register witness v1 " +
+				"spend request without outpoint")
+		}
+
+		// We have an outpoint, so we can set the pkScript to an all
+		// zero Taproot key that we'll compare this spend request to.
+		r.PkScript = ZeroTaprootPkScript
+	}
+
```

Comment:
No, fortunately the `ConfRequest` is the easy case, since we only care about a certain _output_ being confirmed. So if you look at `ConfRequest.MatchesTx()`, we directly compare the `PkScript` of the transaction's _outputs_ with the confirmation request and don't do any `txscript.ComputePkScript()` magic that we do in `SpendRequest.MatchesTx()`.